### PR TITLE
api: refactor, only check cluster existence in middleware

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

https://github.com/tikv/pd/issues/1905 is the source of using context to retrive cluster. Its aim is to eliminate common checks. But, all api handlers that invokes `getCluster` actually have a `*server.Server` or `*server.Handler` in their api handler structs, which means they can get the cluster directly without using context.

As our original aim is to remove redundant checks, we can totally avoid using context, using only middleware, for a consistent way to get extra infos in api handlers: passing `*server.Server` down to handler.

### Check List

Tests

- Unit test
- Integration test

### Release note

- No release note

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]
LGTM. BTW, what do you think about unifying the different API handlers here to merge `h.GetRaftCluster()` and `h.svr.GetRaftCluster()`?

**Comment by xhebox** [comment]
@JmPotato It sounds good to me. I will look into it later.

File: server/api/admin.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func newAdminHandler(svr *server.Server, rd *render.Render) *adminHandler {
 // @Failure 400 {string} string "The input is invalid."
 // @Router /admin/cache/region/{id} [delete]
 func (h *adminHandler) HandleDropCacheRegion(w http.ResponseWriter, r *http.Request) {
-	rc := getCluster(r.Context())
+	rc := h.svr.GetRaftCluster()
```
Comment (rleungx) [inline server/api/admin.go:47]: Do we need to check if `rc != nil`?

**Review by rleungx — APPROVED** [review]

**Pull request merged**